geomesa-convert/geomesa-convert-common/src/test/scala/org/locationtech/geomesa/convert/common/TransformersTest.scala

```diff
@@ -274,7 +274,160 @@ class TransformersTest extends Specification {
+            exp.eval(Array("", "1.2", "1.0", "0.0")) must beFalse
+            exp.eval(Array("", "0.0", "0.0", "0.0")) must beTrue
+          }
+          "float empty string string2float zero default" >> {
+            val exp: Predicate = Transformers.parsePred("floatEq(string2float($1, $3), string2float($2, $3))")
+            exp.eval(Array("", "", "1.0", "0.0")) must beFalse
+            exp.eval(Array("", "", "0.0", "0.0")) must beTrue
+          }
+          "float null string string2float zero default" >> {
+            val exp = Transformers.parsePred("floatEq(string2float($1,$3), string2float($2,$3))")
+            exp.eval(Array("", null, "1.0", "0.0")) must beFalse
+            exp.eval(Array("", null, "0.0", "0.0")) must beTrue
+          }
+          "float string2float null default" >> {
+            val exp: Predicate = Transformers.parsePred("floatEq(string2float($1, $3), string2float($2, $3))")
+            exp.eval(Array("", "1.2", "1.0", null)) must beFalse
+            exp.eval(Array("", "0.0", "0.0", null)) must beTrue
+          }
+          "float empty string string2float null default" >> {
+            val exp: Predicate = Transformers.parsePred("floatEq(string2float($1, $3), string2float($2, $3))")
+            exp.eval(Array("", "", "1.0", null)) must beFalse
+            exp.eval(Array("", "", null, null)) must beTrue
+          }
+          "float null string string2float null default" >> {
+            val exp = Transformers.parsePred("floatEq(string2float($1,$3), string2float($2,$3))")
+            exp.eval(Array("", null, "1.0", null)) must beFalse
+            exp.eval(Array("", null, null, null)) must beTrue
+          }
+        }
+        "string2boolean" >> {
+          "boolean string2boolean true default" >> {
+            val exp = Transformers.parseTransform("string2boolean($1,$2)")
+            exp.eval(Array("", "true", "true")) mustEqual true
+            exp.eval(Array("", "false", "true")) mustEqual false
+            exp.eval(Array("", "", "true")) mustEqual true
+            exp.eval(Array("", null, "true")) mustEqual true
+          }
+          "boolean string2boolean false default" >> {
+            val exp = Transformers.parseTransform("string2boolean($1,$2)")
+            exp.eval(Array("", "true", "false")) mustEqual true
+            exp.eval(Array("", "false", "false")) mustEqual false
+            exp.eval(Array("", "", "false")) mustEqual false
+            exp.eval(Array("", null, "false")) mustEqual false
+          }
+          "boolean string2boolean null default" >> {
+            val exp = Transformers.parseTransform("string2boolean($1,$2)")
+            exp.eval(Array("", "true", null)) mustEqual true
+            exp.eval(Array("", "false", null)) mustEqual false
+            exp.eval(Array("", "", null)) mustEqual null
+            exp.eval(Array("", null, null)) mustEqual null
+          }
+        }
```

Comment:
@drackaer you'd have to put the tests in e.g. geomesa-convert-text, there are examples there you can copy/extend.